lib/quby/answers/services/answer_validator.rb

```diff
@@ -36,10 +36,23 @@ def validate
                   validate_integer(question, validation, value)
                 when :valid_float
                   validate_float(question, validation, value)
+                when :valid_date
+                  value = question.components.each_with_object({}) do |component, hash|
+                    key = question.send("#{component}_key")
+                    hash[component] = answer.send(key)
+                  end
+                  validate_date(question, validation, value)
                 when :regexp
                   validate_regexp(question, validation, value)
                 when :requires_answer
-                  validate_required(question, validation, value)
+                  if question.type == :date
+                    _value = question.answer_keys.each_with_object({}) do |key, hash|
```

Comment:
Dit heb ik gedaan om duidelijk te maken dat het een uitzondering is op hoe validate_... wordt aangeroepen in de andere cases.

Nu hergebruikt de code value, het is wat korter maar ik vind het een beetje hacky aanvoelen.
````ruby
if question.type == :date
                    value = question.answer_keys.each_with_object({}) do |key, hash|
                      hash[key] = answer.send(key)
                    end
                  end
                  validate_required(question, validation, value)
````

Enig idee hoe dit wat consistenter gemaakt kan worden met de rest van de case structuur?